### PR TITLE
Allow for multiple environment configurations.

### PR DESCRIPTION
**Pull request by newswangerd**
Allow users to specify an environment file with `oci-env -e my_custom.env <command>` to launch multiple instances of the developer environment. See the README changes for details.

New environment configurations:
- `SRC_DIR`: location where the git checkouts you want to run from source are. By default this is is set to the directory `oci_env/` is in, but it's no longer required to be set to that. This allows for users to run multiple versions of the environment with different versions of plugins installed.

Breaking Changes for profiles:
- `COMPOSE_CONTEXT` environment variable deprecated. Use `SRC_PATH` for custom compose.yaml files in plugins and `OCI_ENV_DIR` for oci_env
- `DEV_IMAGE_SUFFIX` compose.env setting deprecated. Use `COMPOSE_PROJECT_NAME` to create isolated containers
- `DEV_VOLUME_SUFFIX` compose.env setting deprecated. Use `COMPOSE_PROJECT_NAME` to create isolated volumes

New commands:
- `poll`: Poll the status API until it comes up.
```
$ oci-env poll --help
usage: oci-env poll [-h] [--attempts ATTEMPTS] [--wait WAIT]

optional arguments:
  -h, --help           show this help message and exit
  --attempts ATTEMPTS  Number of attempts to make.
  --wait WAIT          Time in seconds to wait between attempts.
```

Fixes: #27

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,11 +32,11 @@ A developer environment for pulp based off of the [Pulp OCI Images](https://gith
 
     The OCI env project should be in the same directory as any pulp plugins you wish to run.
 
-3. Define your `.compose.env` file.
+3. Define your `compose.env` file.
```
Comment (mdellweg) [inline README.md:35]: :heart_eyes:

File: .github/assets/docker_compose.env [hunk 2]
```diff
@@ -1,3 +1,4 @@
 COMPOSE_PROFILE=_test:_test2
 DEV_SOURCE_PATH=pulpcore:pulp_file
-COMPOSE_BINARY=docker
\ No newline at end of file
+COMPOSE_BINARY=docker
+SECOND_SERVICE_PORT=8002
```
Comment (bmbouter) [inline .github/assets/docker_compose.env:4]: What is this for, the "second one started"?

File: README.md [hunk 3]
```diff
@@ -35,11 +35,11 @@ A developer environment for pulp based off of the [Pulp OCI Images](https://gith
     Note, the `/src/` folder in the container is the parent folder containing the `oci_env` and all
```
Comment (bmbouter) [inline README.md:35]: I think this path is changing right? Can you grep this file for any other updates? Apologies if I'm misunderstanding things here.

File: .github/assets/docker_compose.env (hunk 2 above)
Comment (newswangerd) [reply .github/assets/docker_compose.env:4]: That's just for CI. I created a set of profiles for CI, and one of them runs a web server to verify that other services can be added to the system: https://github.com/pulp/oci_env/pull/31/files#diff-50bc831b758ad43f28642443148d9d1f6fc62cce155af3000c4f16d0dbdae959

Originally it just listened to port 8002 by default, but you can't have two containers on the same port, so I had to add a configuration for that.

File: README.md (hunk 3 above)
Comment (newswangerd) [reply README.md:35]: The path in the container stays the same. The only thing that's changing is the path that gets mounted into the container. `/src/` in the container used to always map to `oci_env/../` on the host machine. Now it maps to `oci_env/../` by default, but you can also configure it by setting `SRC_PATH` in your .env configuration if you want to change it. This allows you to have one checkout for `oci_env` that lets you run multiple different `SRC_PATH`s. For example I can check out `oci_env` in `~/code/oci_env` and create another directory at `~/code/hub/` that has different versions of galaxy_ng checked out on different branches:

```
~/code/hub
├── env-insights
│   ├── ansible-hub-ui
│   └── galaxy_ng
├── env-standalone
│   ├── ansible-hub-ui
│   └── galaxy_ng
├── insights.compose.env
└── standalone.compose.env
```

In this example I have `SRC_DIR=~/code/hub/env-insights/` set in my `insights.compose.env` file and `SRC_DIR=~/code/hub/env-standalone/` in my `standalone.compose.env`. I was using this to test syncing between different versions of hub

**Review by dkliban — APPROVED** [review]

**Pull request merged**